app/models/stamp.rb

```diff
@@ -17,4 +19,13 @@ def deactivated?
   def deactivate!
     update_attributes!(deactivated_at: DateTime.now) if active?
   end
+
+  private
+
+  # Name must be unique within the scope of active stamps
+  def validate_name_active_uniqueness
+    if Stamp.where(name: name, deactivated_at: nil).any? { |s| s.id != id }
```

Comment:
Instead of looping through all active Stamps checking it's not the one being validated, you could query the db to check for the existence of an active one with the same name i.e.

`Stamp.where("name = ? AND id != ? AND deactivated_at is NULL", name, id).exists?`